src/mission_start.cpp

```diff
@@ -359,18 +359,16 @@ void mission_start::kill_100_z( mission *miss )
 {
     npc *p = g->find_npc( miss->npc_id );
     p->set_attitude( NPCATT_FOLLOW );//npc joins you
-    int killed = 0;
-    killed += g->kill_count( miss->monster_species );
-    miss->monster_kill_goal = 100 + killed; //your kill score must increase by 100
+    //number of monsters from given species you killed at the mission start plus number required to kill
+    miss->monster_killed = g->kill_count( miss->monster_species ) + miss->monster_kill_goal;
```

Comment:
The documentation (and the code) imply that this is the number of kills one *has* to achieve (in the future). It's not the number of kills one has already achieved (it's not the number of "killed" monsters).

The comparison in `` does not make much sense either: `kill_count( ... ) >= monster_killed` - isn't `kill_count` the number of "killed monsters"? It reads as "number of kills of type ... >= number of monsters killed".

The documentation is also difficult to make sense of:

`monster_kill_goal; // The kill count you wish to reach` (it's not compared to `g->kill_count`, which seems logical as it explicitly says "kill count" in the comment.)

`int monster_killed; // The kill count at the mission start plus monster_kill_goal` This would make it the "kill count you need to reach to succeed the mission".
         
Suggestion: `kill_count_to_reach`. And one does probably not need to store a copy of the kill count goal from the mission type in the mission instance. One can look up that value any time in the linked `mission_type` object and it won't be changed through out the time of the mission.